core/src/main/scala/org/apache/spark/sql/SnappyStrategies.scala

```diff
@@ -81,13 +81,20 @@ private[sql] trait SnappyStrategies {
       case _ => Nil
     }
 
+    def isLocalJoin(plan: LogicalPlan): Boolean = plan match {
+      case ExtractEquiJoinKeys(joinType, _, _, _, left, right) =>
+        (canBuildRight(joinType) && canLocalJoin(right)) ||
+            (canBuildLeft(joinType) && canLocalJoin(left))
```

Comment:
These conditions only check for replicated tables. canBuildLeft(joinType) && canBuildLocalHashMap(left) || canBuildRight(joinType) && canBuildLocalHashMap(right) checks for non-replicated table joins as well.